skbio/io/format/embl.py

```diff
@@ -0,0 +1,1497 @@
+    return output
+
+
+# Method to determine if file is in EMBL format or not. A uniprot embl format
+# can't be parsed by this module (at the moment)
+@embl.sniffer()
+def _embl_sniffer(fh):
+    try:
+        line = next(_line_generator(fh, skip_blanks=True, strip=False))
+    except StopIteration:
+        return False, {}
+
+    try:
+        _parse_id([line])
+
+    except EMBLFormatError:
+        return False, {}
+
+    return True, {}
+
+
+@embl.reader(None)
+def _embl_to_generator(fh, constructor=None, **kwargs):
+    for record in _parse_embls(fh):
+        yield _construct(record, constructor, **kwargs)
+
+
+# Method to read EMBL data as skbio.sequence.DNA
+@embl.reader(Sequence)
+def _embl_to_sequence(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_embls(fh), seq_num)
+    return _construct(record, Sequence, **kwargs)
+
+
+# Method to read EMBL data as skbio.sequence.DNA
+@embl.reader(DNA)
+def _embl_to_dna(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_embls(fh), seq_num)
+    return _construct(record, DNA, **kwargs)
+
+
+# Method to read EMBL data as skbio.sequence.DNA
+@embl.reader(RNA)
+def _embl_to_rna(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_embls(fh), seq_num)
+    return _construct(record, RNA, **kwargs)
+
+
+# No protein support at the moment
+@embl.reader(Protein)
+def _embl_to_protein(fh, seq_num=1, **kwargs):
+    # no protein support, at the moment
+    # record = _get_nth_sequence(_parse_embls(fh), seq_num)
+    # return _construct(record, Protein, **kwargs)
+    raise EMBLFormatError("There's no protein support for EMBL record")
+
+
+# Writer methods
+@embl.writer(None)
+def _generator_to_embl(obj, fh):
+    for obj_i in obj:
+        _serialize_single_embl(obj_i, fh)
+
+
+@embl.writer(Sequence)
+def _sequence_to_embl(obj, fh):
+    _serialize_single_embl(obj, fh)
+
+
+@embl.writer(DNA)
+def _dna_to_embl(obj, fh):
+    _serialize_single_embl(obj, fh)
+
+
+@embl.writer(RNA)
+def _rna_to_embl(obj, fh):
+    _serialize_single_embl(obj, fh)
+
+
+@embl.writer(Protein)
+def _protein_to_embl(obj, fh):
+    # no protein support, at the moment
+    # _serialize_single_embl(obj, fh)
+    raise EMBLFormatError("There's no protein support for EMBL record")
+
+
+def _construct(record, constructor=None, **kwargs):
+    '''Construct the object of Sequence, DNA, RNA, or Protein.'''
+
+    # sequence, metadata and interval metadata
+    seq, md, imd = record
+
+    if 'lowercase' not in kwargs:
+        kwargs['lowercase'] = True
+
+    if constructor is None:
+        unit = md['LOCUS']['unit']
+        if unit == 'bp':
+            # RNA mol type has T instead of U for genbank from from NCBI
+            constructor = DNA
+
+        elif unit == 'aa':
+            # no protein support, at the moment
+            # constructor = Protein
+            raise EMBLFormatError("There's no protein support for EMBL record")
+
+    if constructor == RNA:
+        return DNA(
```

Comment:
Embl store rRNA data as DNA (ex [LK021130.1:74067..75610:rRNA](http://www.ebi.ac.uk/ena/data/view/Non-coding:LK021130.1:74067..75610:rRNA&display=text)). Sorry, you can instantiate a RNA object directly but it will be converted in DNA in writing, for example:
```python
import io
import skbio

# load test object for simplicity
from skbio.io.format.tests.test_embl import ReaderTests

test = ReaderTests()
test.setUp()

# this tuple contains, a RNA sequence, metadata, IntervalMetadata, and RNA class
seq, md, imd, constructor = test.single_rna

# even if constructor is skbio.sequence._rna.RNA, do
obs = skbio.RNA(seq, metadata=md, lowercase=True, interval_metadata=imd)

# obj is skbio.sequence._rna.RNA
type(obs)

# Write this RNA obj into file
handle = io.StringIO()
skbio.io.write(obs, format="embl", into=handle)

# but data is written in DNA
print(handle.getvalue())
```
genbank module works in a similat way (tests and code for embl were inspired from test and code of genbank)